### PR TITLE
cleanup the manifest apis

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug
Fixes: None, related to the licenses work
Regression: No  
If Regression then when did it last work:   
If Regression then how are we preventing it in future:   

## Fix
Details: Cleaning up redundant APIs. 
The manfiest validation cannot/should not be skipped. 

## Testing/Validation
Tests Added: No
Reason for not adding tests:  
Validation done:

### CODE REVIEW

File: src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/Manifest.cs [hunk 1]
```diff
@@ -66,23 +66,10 @@ public void Save(Stream stream)
...
-        }
 
-        public void Save(Stream stream, bool validate, int minimumManifestVersion)
-        {
-            if (validate)
-            {
-                // Validate before saving
-                Validate(this);
-            }
+            Validate(this);
```
Comment (PatoBeltran) [inline src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/Manifest.cs:70]: why make it mandatory that all saves validate?

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/Manifest.cs:70]: This is manifest metadata validation. 
Namely validation such as, make sure the version is correct, amke sure the ID is valid etc. 

There's also no place in the code that currently skips the validation. 
It's unreachable.

**Review by PatoBeltran — APPROVED** [review]

File: src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/Manifest.cs (hunk 1 above)
Comment (PatoBeltran) [reply src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/Manifest.cs:70]: I'm all for deleting unnecessary code. 💯

**Comment by ericstj** [comment]
@rrelyea this was a binary breaking change between minor versions of NuGet.

**Pull request merged**